.husky/pre-commit

```diff
@@ -7,4 +8,27 @@ npm run typecheck
 npm run update:toc
 npm run check:pom
 
-git add .
+# Check docs/src directory for TypeScript and ESLint errors
+echo "Checking docs/src directory..."
+
+# Verify docs directory exists
+if [ ! -d "docs" ]; then
+  echo "⚠️  docs directory not found, skipping docs checks"
+else
+  # Use subshell to ensure directory is reset even on failure
+  (
+    cd docs || exit 1
+    
+    if ! npx tsc --noEmit; then
+      echo "❌ TypeScript check failed in docs/src"
```

Comment:
The existing check should recursively check the `docs/` directory. This addition seems unnecessary.